crates/ruff_linter/src/rules/flake8_bandit/rules/hardcoded_sql_expression.rs

```diff
@@ -1,18 +1,26 @@
 use once_cell::sync::Lazy;
 use regex::Regex;
-use ruff_python_ast::{self as ast, Expr, Operator};
-
 use ruff_diagnostics::{Diagnostic, Violation};
 use ruff_macros::{derive_message_formats, violation};
 use ruff_python_ast::str::raw_contents;
+use ruff_python_ast::{self as ast, Expr, Operator};
 use ruff_source_file::Locator;
 use ruff_text_size::Ranged;
 
 use crate::checkers::ast::Checker;
 
+// Implementation of regex from bandit
 static SQL_REGEX: Lazy<Regex> = Lazy::new(|| {
-    Regex::new(r"(?i)\b(select\s.+\sfrom\s|delete\s+from\s|(insert|replace)\s.+\svalues\s|update\s.+\sset\s)")
-        .unwrap()
+    Regex::new(
+        r"(?ix)
+        (
+            select\s+.*from\s |
+            delete\s+from\s  |
+            insert\s+into\s.*values\s |
+            update\s+.*set\s
+        )",
+    )
+    .unwrap()
```

Comment:
The regex seems to be missing a few things. Sorry for a lot of back and forth but we should revert it back to the original regex and only update the relevant parts that are required to fix the linked issue.

1. The `select` should be `select\s+.*\s+from\s` (currently it doesn't account for multiple whitespaces)
2. The `replace` is missing in the updated regex, so it should be `(insert|replace)\s+.*\s+values\s`
3. The `update` should be `update\s+.*\s+set\s` so we use the similar pattern as others for whitespaces (`\s+.*\s+`)

Reply:
With the updated regex, it matches all the examples that you've specified in https://github.com/astral-sh/ruff/pull/13574#issuecomment-2414889325 except for the last one which contains a line continuation character.

Reply:
Yes, this resulted in 57 matches. Trying to figure out the reason for missing 
```py
query58 = f"SELECT\
 * FROM {var}.table"
```

Reply:
I would be fine landing this PR without line continuation support because I understand that line continuation weren't supported before. We can create a separate PR to add line continuation support. Wdyt?

Reply:
Yes, it does make sense. 
Line continuation was a part of additional test cases only.